libvcx/src/libindy/utils/wallet.rs

```diff
@@ -49,6 +60,43 @@ pub fn reset_wallet_handle() {
     agency_wallet::reset_wallet_handle();
 }
 
+pub fn create_wallet_from_config(config: &str) -> VcxResult<()> {
+    let config: WalletConfig = serde_json::from_str(config)
+        .map_err(|err| VcxError::from_msg(VcxErrorKind::InvalidJson, format!("Cannot deserialize WalletConfig {:?}, err: {:?}", config, err)))?;
+
+    let wh = create_and_open_as_main_wallet(
+        &config.wallet_name,
+        &config.wallet_key,
+        &config.wallet_key_derivation,
+        config.wallet_type.as_ref().map(String::as_str),
+        config.storage_config.as_ref().map(String::as_str),
+        config.storage_credentials.as_ref().map(String::as_str),
+    )?;
+    set_wallet_handle(wh);
+    trace!("initialized wallet");
+
+    // If MS is already in wallet then just continue
+    anoncreds::libindy_prover_create_master_secret(settings::DEFAULT_LINK_SECRET_ALIAS).ok();
+    
+    close_main_wallet()?;
+    Ok(())
+}
+
+pub fn configure_issuer_wallet(enterprise_seed: &str, institution_name: &str) -> VcxResult<String> {
```

Comment:
I think this should only take the seed, as this function is now doing 2 things:
- setting up the wallet
- initializing memory

And actually, `institution name` is used by verifier, but mobile apps might commonly be (going bit into future) as well in position of verifier.`

Reply:
I don't think the first one is an argument - it will set up the wallet and initialize settings anyway. As for the second one, it is true that this will probably be the case in the future, but making the initialization even more complex introducing another method just to set up the config for a future use case seems unnecessary for now. I would agree to to rename to `configure_institution_wallet`, however.

Reply:
`institution_name ` could be just part of config being passed to `vcx_init_core`. if `configure_issuer_wallet` would be the only way to configure `institution_name` in library memory settings, that would be an agent would have to actually on every startup  call `configure_issuer_wallet `. But `configure_issuer_wallet ` should be only 1 time operation per wallet.